dev/install-common-deps.sh

```diff
@@ -28,7 +28,8 @@ conda create -q -n test-environment python=3.6
 source activate test-environment
 
 python --version
-pip install --upgrade pip==19.3.1
+pip install --upgrade pip
+pip --version
```

Comment:
Confirmed the pip version has been updated:

```
++ pip --version
pip 21.0.1 from /usr/share/miniconda/envs/test-environment/lib/python3.6/site-packages/pip (python 3.6)
```

https://github.com/mlflow/mlflow/runs/1860085508#step:4:245

Reply:
And all the CI checks have passed :)